cooperative_lanechange/src/cooperative_lanechange.cpp

```diff
@@ -134,22 +137,40 @@ namespace cooperative_lanechange
             return true;
         }
         lanelet::ConstLanelet current_lanelet = current_lanelets[0].second;
-
+        ROS_DEBUG_STREAM("current llt id " << current_lanelet.id());
+        
         //Create temporary route between the two vehicles
         lanelet::ConstLanelet start_lanelet = veh2_lanelet;
         lanelet::ConstLanelet end_lanelet = current_lanelet;
         lanelet::traffic_rules::TrafficRulesUPtr traffic_rules = lanelet::traffic_rules::TrafficRulesFactory::create(lanelet::Locations::Germany,lanelet::Participants::VehicleCar);
+        ROS_DEBUG_STREAM("traffic rules created]");
+        
         lanelet::routing::RoutingGraphUPtr map_graph = lanelet::routing::RoutingGraph::build(*(wm_->getMap()), *traffic_rules);
+        ROS_DEBUG_STREAM("Graph created]");
 
         auto temp_route = map_graph->getRoute(start_lanelet, end_lanelet);
+        ROS_DEBUG_STREAM("Route created]");
         
         auto shortest_path2 = temp_route.get().shortestPath();
+        ROS_DEBUG_STREAM("Shorted path created size: " << shortest_path2.size());
+        for (auto llt : shortest_path2)
+        {
+            ROS_DEBUG_STREAM("llt id  route: " << llt.id());
+        }  
+
 
         //To find downtrack- distance of veh2 from current to start of end lanelet + downtrack of veh1 in its own current lanelet
-        lanelet::BasicPoint2d p = shortest_path2.back().centerline2d().front(); //Start point of last lanelet in path
         
-        double downtrack_1 = wm_->routeTrackPos(p).downtrack;
-        double remaining_downtrack = downtrack_1 - veh2_downtrack + ego_current_downtrack;
+        double veh2_lanelet_start_downtrack = wm_->routeTrackPos(shortest_path2.front().centerline2d().front()).downtrack;
+        ROS_DEBUG_STREAM("veh2_lanelet_start_downtrack:" << veh2_lanelet_start_downtrack);
+
+        double ego_current_downtrack = wm_->routeTrackPos(ego_pos).downtrack;      
+        ROS_DEBUG_STREAM("ego_current_downtrack:" << ego_current_downtrack);
+        
+        
+        double remaining_downtrack = ego_current_downtrack - veh2_downtrack - veh2_lanelet_start_downtrack;
```

Comment:
This whole block here is kind of confusing, both in it's purpose and it's naming. `remaining downtrack` implies we're using some of it up? Are both of these downtracks in lanelet-reference frame or just veh2? Do we need to be in the same lanelet as veh2 for this to hold? Some comments and clearer variable names could help out here.

Reply:
In this part we are creating a temporary route from the veh2 to veh1 and using the shortest path from that, to calculate the downtrack distance between the two vehicles.
This downtrack distance is used as the gap between the two.
I'll add comments and rename the variables